include/vm_map.h

```diff
@@ -26,7 +27,9 @@ typedef struct vm_map {
   TAILQ_HEAD(vm_map_list, vm_map_entry) list;
   SPLAY_HEAD(vm_map_tree, vm_map_entry) tree;
   size_t nentries;
-  pmap_t *pmap;
+  pmap_t *const pmap;
```

Comment:
What do we gain by declaring this pointer as `const`? To circumvent constness you use `*((pmap_t **)(&x)) = p;` assignment, which looks pretty ugly.